src/api/Metadata.js

```diff
@@ -154,26 +176,223 @@ class Metadata extends File {
+                if (template) {
+                    editors.push(this.createEditor(instance, template, canEdit));
+                }
             }),
-            'data.entries',
-            [],
         );
+        return editors;
+    }
+
+    /**
+     * API for getting metadata editors
+     *
+     * @param {string} fileId - Box file id
+     * @param {Function} successCallback - Success callback
+     * @param {Function} errorCallback - Error callback
+     * @param {boolean} hasMetadataFeature - metadata feature check
+     * @param {Object} options - fetch options
+     * @return {Promise}
+     */
+    async getMetadata(
+        file: BoxItem,
+        successCallback: ({ editors: Array<MetadataEditor>, templates: Array<MetadataEditorTemplate> }) => void,
+        errorCallback: ElementsErrorCallback,
+        hasMetadataFeature: boolean,
+        options: FetchOptions = {},
+    ): Promise<void> {
+        const { id, permissions, is_externally_owned }: BoxItem = file;
+        this.errorCode = ERROR_CODE_FETCH_METADATA;
+        this.successCallback = successCallback;
+        this.errorCallback = errorCallback;
+
+        // Check for valid file object.
+        // Need to eventually check for upload permission.
+        if (!id || !permissions) {
+            this.errorHandler(getBadItemError());
+            return;
+        }
+
+        const cache: APICache = this.getCache();
+        const key = this.getMetadataCacheKey(id);
+
+        // Clear the cache if needed
+        if (options.forceFetch) {
+            cache.unset(key);
+        }
+
+        // Return the cached value if it exists
+        if (cache.has(key)) {
+            this.successHandler(cache.get(key));
+            if (!options.refreshCache) {
+                return;
+            }
+        }
+
+        try {
+            const customPropertiesTemplate: MetadataEditorTemplate = this.getCustomPropertiesTemplate();
+            const [instances, globalTemplates, enterpriseTemplates] = await Promise.all([
+                this.getInstances(id),
+                this.getTemplates(id, METADATA_SCOPE_GLOBAL),
+                hasMetadataFeature ? this.getTemplates(id, METADATA_SCOPE_ENTERPRISE) : Promise.resolve([]),
+            ]);
+
+            const editors = await this.getEditors(
+                id,
+                instances,
+                customPropertiesTemplate,
+                enterpriseTemplates,
+                globalTemplates,
+                !!permissions.can_upload,
+            );
+
+            if (this.isDestroyed()) {
+                return;
+            }
+
+            const metadata = {
+                editors,
+                templates: this.getUserAddableTemplates(
+                    customPropertiesTemplate,
+                    enterpriseTemplates,
+                    hasMetadataFeature,
+                    is_externally_owned,
+                ),
+            };
+            cache.set(key, metadata);
```

Comment:
I think this might be done elsewhere in our codebase, but if we have already fetched the data we should at least set the cache, but not call the callbacks